src/main/cpp/util/profiler.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/main/cpp/util/profiler.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+
+namespace blaze_util {
+namespace profiler {
+
+Task::~Task() {
+  Duration duration = GetDuration();
+  printf("Task: %s: %" PRIu64 " calls (%" PRId64 " us)\n", name_, GetCalls(),
+         duration.micros_);
+}
+
+void StopWatch::PrintAndReset(const char* name) {
+  Duration elapsed_us = ElapsedTime();
+  printf("StopWatch: %s: %" PRId64 " us elapsed\n", name, elapsed_us.micros_);
```

Comment:
nit: either use BAZEL_LOG or print to stderr instead (also everywhere else)

Reply:
Done -- using fprintf on stderr now.

Tried using BAZEL_LOG in `Task`'s destructor but got a segfault. I haven't dug into the logger's code but I'm supposing it's shut down before the Task object is destructed.